internal/olm/operator/bundleupgrade/upgrade.go

```diff
@@ -86,6 +85,7 @@ func (u *Upgrade) setup(ctx context.Context) error {
 	u.IndexImageCatalogCreator.PackageName = u.OperatorInstaller.PackageName
 	u.IndexImageCatalogCreator.BundleImage = u.BundleImage
 	u.IndexImageCatalogCreator.IndexImage = registry.DefaultIndexImage
+	u.IndexImageCatalogCreator.ChannelName = strings.Split(labels[registrybundle.ChannelsLabel], ",")[0]
```

Comment:
I think we should be doing the same thing here as we did for run bundle, which is to handle scenarios where channel metadata doesn't exist on the bundle. So something like the below:

```suggestion
	if _, hasChannelMetadata := labels[registrybundle.ChannelsLabel]; hasChannelMetadata {
		u.ChannelName = strings.Split(labels[registrybundle.ChannelsLabel], ",")[0]
	} else {
		u.ChannelName = defaultChannel
	}
```